src/Purestone/Game.hs

```diff
@@ -26,19 +26,19 @@ processMove = undefined
 -- | `draw` takes a player and draws the top card from their deck into their hand
 draw :: Player -> Player
 -- If no cards, the player takes damage equal to amount of overdraws
-draw (Player hp h [] c m bs dp o) = (Player (hp - (o+1)) h [] c m bs dp (o+1))
+draw (Player hp h [] c m bs dp o) = Player (hp - (o+1)) h [] c m bs dp (o+1)
 draw (Player hp h (x:xs) c m bs dp o) = case length h of
-  10 -> (Player hp h xs c m bs dp o) -- If the player has 10 cards, drawn card is lost
-  otherwise -> (Player hp (x:h) xs c m bs dp o)
+  10 -> Player hp h xs c m bs dp o -- If the player has 10 cards, drawn card is lost
+  otherwise -> Player hp (x:h) xs c m bs dp o
```

Comment:
Could this be implemented more robustly (and elegantly) as e.g. something like:
```haskell
draw player@Player{ playerDeck = x:xs, ..} 
    | length playerHand >= 10 = player{ playerDeck = xs }
    | otherwise = player{ playerHand = x : playerHand, playerDeck = xs  }
```